test_util/helpers.py

```diff
@@ -4,10 +4,34 @@
 import requests
 import retrying
 
+import test_util
+
 Host = namedtuple('Host', ['private_ip', 'public_ip'])
 SshInfo = namedtuple('SshInfo', ['user', 'home_dir'])
 
 
+class DcosUser:
+    """A lightweight user representation."""
+    def __init__(self, auth_json):
+        self.auth_json = auth_json
+        self.auth_header = None
+        self.auth_token = None
+        self.auth_cookie = None
+
+    def authenticate(self, cluster):
+        assert isinstance(cluster, test_util.cluster_api.ClusterApi), 'Unrecognized cluster object!'
+        logging.info('Attempting authentication')
+        r = cluster.post(path='/acs/api/v1/auth/login', disable_suauth=True, payload=self.auth_json)
+        if r.status_code >= 400:
+            return False
+        assert r.ok, 'Authentication failed with status_code: {}'.format(r.status_code)
+        self.auth_token = r.json()['token']
```

Comment:
would be handy to log / print r.json() before this. If we got an empty json object, this would just break hard and be complicated to debug.